File: Find Closest Number to Zero - Leetcode 2239/Find Closest Number to Zero - Leetcode 2239.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findClosestNumber = function(nums) {
    let closest = nums[0];
    
    for (let x of nums) {
        if (Math.abs(x) < Math.abs(closest)) {
            closest = x;
        }
    }
    
    if (closest < 0 && nums.includes(Math.abs(closest))) {
        return Math.abs(closest);
    } else {
        return closest;
    }
};

File: Find Closest Number to Zero - Leetcode 2239/Find Closest Number to Zero - Leetcode 2239.rs
fn abs(x: i32) -> i32 {
    if x < 0 {
       return -x;
    }
    x
}

fn find_closest(nums: Vec<i32>) -> i32 {
    let mut closest = nums[0];
    for i in 0..nums.len() {
        if abs(nums[i]) < abs(closest) {
            closest = nums[i];
        } else if  abs(nums[i]) == abs(closest) && nums[i] > closest {
            closest = nums[i];
        }
    }
    closest
}